.github/workflows/build-and-test.yml

```diff
@@ -8,64 +8,30 @@ on:
 
 jobs:
   build-and-test-osx:
-    runs-on: macos-latest
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash
     strategy:
+      fail-fast: false
       matrix:
-        node-version: [16.x]
+        node-version: [16,18,20]
+        os: [macos-latest,ubuntu-latest,windows-latest]
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node-version }}
       - run: script/ci/build-and-test.sh
+        if: runner.os != 'Windows'
         env:
           NODE_VERSION: ${{ matrix.node-version }}
           PACT_BROKER_FEATURES: publish_pacts_using_old_api
-
-  build-and-test-ubuntu:
-    runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        node-version: [16.x]
-
-    steps:
-      - uses: actions/checkout@v2
-      - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v1
-        with:
-          node-version: ${{ matrix.node-version }}
-      - name: Set MSVS version
-        run: npm config set msvs_version 2017
       - run: script/ci/build-and-test.sh
```

Comment:
This needs to be `bash script/ci/build-and-test.sh`, otherwise it won't work on windows.

Reply:
you can set it at a defaults level at the top level of a job, so it applies to all run commands 👍🏾 saves adding bash to every step because windows

Reply:
Hm ok. Last time I looked at this, Windows needed both:

```
run: bash script/ci/build-and-test.sh
shell: bash
```

but perhaps it has improved.

Reply:
Ahh I've only ever needed `shell: bash` - I use `bash` prefix when I use powershell or command prompt in windows, although it tends to get annoyed and I have to do `bash -c 'annoying folder with spaces'`

because `Program Files` I mean seriously dudes